### PR TITLE
(IAC-422) Update EKS Module

### PR DESCRIPTION
**Pull request by jarpat**

### CODE REVIEW

**Comment by thpang** [comment]
@jarpat thanks for creating this. I'll monitor and work from here as well when time permits.

File: main.tf [hunk 1]
```diff
@@ -95,10 +97,8 @@ module "eks" {
   ################################################################################
   # Cluster Security Group
   ################################################################################
-  create_cluster_security_group                  = true                            # v17: cluster_create_security_group
-  # cluster_security_group_id                      = local.cluster_security_group_id
-  # cluster_security_group_name                    = 
-
+  create_cluster_security_group                  = false  # v17: cluster_create_security_group
```
[the remark this answers is not in the document]
Comment (thpang) [reply main.tf:100]: I can't tell here from all the comments and active work what was in the original code base. So there will need to be some clarification for these changes.

**Comment by manoatsas** [comment]
Question: should these variables with '_workers_\_*' prefix be renamed to '_nodes_\_*' to match AWS/EKS module naming conventions - https://github.com/sassoftware/viya4-iac-aws/blob/update_eks_module/variables.tf#L332? That would be a breaking change. If these can be renamed then all internal references can be switched to use '_workers_\_*' too.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG TERRAFORM_VERSION=1.1.6
+ARG TERRAFORM_VERSION=1.0.0
```
Comment (thpang) [inline Dockerfile:1]: If this is being changed, is the doc being updated as well?

File: README.md [hunk 3]
```diff
@@ -44,7 +44,7 @@ The following are also required:
   
 #### Terraform Requirements:
 
-- [Terraform](https://www.terraform.io/downloads.html) v1.0.0
+- [Terraform](https://www.terraform.io/downloads.html) v1.1.6
```
Comment (thpang) [inline README.md:47]: This needs to match the code.

File: locals.tf [hunk 4]
```diff
@@ -31,47 +30,93 @@ locals {
...
+          }
+        }
+      }
+      desired_size                      = var.default_nodepool_node_count
+      min_size                          = var.default_nodepool_min_nodes
+      max_size                          = var.default_nodepool_max_nodes
+      taints                            = { for i, taint in var.default_nodepool_taints : "default-${i}"=> { 
+                                              "key" = split("=", taint)[0], 
+                                              "value"= split(":", split("=", taint)[1])[0], 
+                                              "effect"=length(regexall(":No", taint)) > 0 ? upper(replace(split(":", split("=", taint)[1])[1], "No", "NO_")) : upper(replace(split(":", split("=", taint)[1])[1], "No", "_NO_"))
```
Comment (thpang) [inline locals.tf:54]: What is this new effect items used for? And why the No filters and adjustments on the taints?

File: README.md (hunk 3 above)
Comment (thpang) [inline README.md:47]: is this correct? I just saw something where you were switching back to Terraform 1.0.0

File: locals.tf [hunk 5]
```diff
@@ -31,47 +30,93 @@ locals {
...
-      asg_min_size                         = var.default_nodepool_min_nodes
-      asg_max_size                         = var.default_nodepool_max_nodes
-      kubelet_extra_args                   = "--node-labels=${replace(replace(jsonencode(var.default_nodepool_labels), "/[\"\\{\\}]/", ""), ":", "=")} --register-with-taints=${join(",", var.default_nodepool_taints)}"
-      additional_userdata                  = (var.default_nodepool_custom_data != "" ? file(var.default_nodepool_custom_data) : "")
-      metadata_http_endpoint               = var.default_nodepool_metadata_http_endpoint
-      metadata_http_tokens                 = var.default_nodepool_metadata_http_tokens
-      metadata_http_put_response_hop_limit = var.default_nodepool_metadata_http_put_response_hop_limit
-
+  # Mapping node_pools to node_groups
+  default_node_pool = {
```
Comment (thpang) [inline locals.tf:34]: You've changed how things are defined here. Going from lists to objects. Need to understand what/why this is needed.

File: locals.tf [hunk 6]
```diff
@@ -31,47 +30,93 @@ locals {
...
+      }
+      # Launch Template
+      create_launch_template          = true
+      launch_template_name            = "${local.cluster_name}-default-lt"
+      launch_template_use_name_prefix = true
+      tags                            = var.autoscaling_enabled ? merge(var.tags, { key = "k8s.io/cluster-autoscaler/${local.cluster_name}", value = "owned", propagate_at_launch = true }, { key = "k8s.io/cluster-autoscaler/enabled", value = "true", propagate_at_launch = true}) : var.tags
     }
-  ]
+  }
+  default_node_group = {}
```
Comment (thpang) [inline locals.tf:73]: What's this for?

File: outputs.tf [hunk 7]
```diff
@@ -8,9 +8,9 @@ output "kube_config" {
   sensitive = true
 }
 
-output "worker_iam_role_arn" {
-  value = module.eks.worker_iam_role_arn
-}
+# output "worker_iam_role_arn" {
+#   value = module.eks.worker_iam_role_arn
+# }
```
Comment (thpang) [inline outputs.tf:13]: Has this been completely removed?

**Review by thpang — CHANGES_REQUESTED** [review]
Questions and items that need review. These will probably be answered in the code review meeting.

File: locals.tf (hunk 4 above)
Comment (manoatsas) [reply locals.tf:54]: See AWS docs https://docs.aws.amazon.com/eks/latest/userguide/node-taints-managed-node-groups.html

File: locals.tf (hunk 6 above)
Comment (manoatsas) [reply locals.tf:73]: Good catch 👍🏼 Removed it.

File: outputs.tf (hunk 7 above)
Comment (manoatsas) [reply outputs.tf:13]: Yes no longer supported by EKS module

File: README.md (hunk 3 above)
Comment (manoatsas) [reply README.md:47]: Fixed it back to 1.0.0

File: Dockerfile (hunk 2 above)
Comment (manoatsas) [reply Dockerfile:1]: No longer applicable

File: locals.tf (hunk 5 above)
Comment (manoatsas) [reply locals.tf:34]: Types change is required by EKS module, see - https://github.com/terraform-aws-modules/terraform-aws-eks/releases/tag/v18.0.0

File: locals.tf (hunk 4 above)
Comment (thpang) [reply locals.tf:54]: Might want to add the link to references to these items in the code base so others are aware.

Comment (manoatsas) [reply locals.tf:54]: https://github.com/sassoftware/viya4-iac-aws/blob/update_eks_module/locals.tf#L93

**Review by thpang — COMMENTED** [review]
These changes look good.

**Review by thpang — APPROVED** [review]

**Comment by jarpat** [comment]
Looks good.

**Pull request merged**